bpfassets/perf_event/perf_event.c

```diff
@@ -192,6 +197,19 @@ static inline u64 get_on_cpu_avg_freq(u32 *cpu_id, u64 on_cpu_cycles_delta, u64
 // int kprobe__finish_task_switch(switch_args *ctx)
 int kprobe__finish_task_switch(struct pt_regs *ctx, struct task_struct *prev)
 {
+    u32 initial = SAMPLE_RATE, *sample_counter_value, sample_counter_key = 1234;
+    // only do sampling if sample rate is set
+    if (initial != 0) {
+        sample_counter_value = sample_rate.lookup_or_try_init(&sample_counter_key, &initial);
+        if (sample_counter_value > 0) {
+            if (*sample_counter_value > 0) {
+                (*sample_counter_value)--;
+                return 0;
+            }
+        }
```

Comment:
New to this so please excuse if this suggestion looks stupid . Would this work? 


```suggestion
        if (sample_counter_value  && *sample_counter_value > 0) {
                (*sample_counter_value)--;
                return 0;
        }
```